plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java

```diff
@@ -881,12 +879,9 @@ public int compare(Viewer viewer, Object e1, Object e2) {
                 return cat1 - cat2;
             }
 
-            if (e1 instanceof DBNDatabaseNode n1 && e2 instanceof DBNDatabaseNode n2) {
-                Object o1 = n1.getValueObject();
-                Object o2 = n2.getValueObject();
-                if (o1 instanceof DBPObjectWithOrdinalPosition c1 && o2 instanceof DBPObjectWithOrdinalPosition c2) {
-                    return Integer.compare(c1.getOrdinalPosition(), c2.getOrdinalPosition());
-                }
+            Object input = viewer.getInput();
+            if (input instanceof Collection<?> list && !list.contains(e1) && !list.contains(e2)) {
```

Comment:
I don't understand the meaning behind this logic. What does it do? Why do we operate on the viewer's input and completely ignore the arguments we are supposed to compare?

Reply:
I suppose it was Serge's suggestion to apply the comparator only for top-level nodes